text/0000-std-rpc-api.md

```diff
@@ -0,0 +1,503 @@
+
+As expected, RPC signature therefore its hash will be changed if assembly, return type, enclosing type, method name and/or any method param type changes (but names of method parameters can be changed as they are not a part of the method signature).
+
+A change in the RPC signature will lead into a different send/receive codepath with different serialization code and execute a different method body. Previous versions of the RPC method will not be executed by the new RPC method with the new signature.
+
+### Cross-Build Compatibility ✅
+
+As long as the RPC method signature is kept the same, it will be compatible between different builds.
+
+### Cross-Version Compatibility ✅
+
+As long as the RPC method signature is kept the same, it will be compatible between different versions.
+
+### Cross-Project Compatibility ❌
+
+Since project name or any project-specific token is not being a part of RPC signature, it is possible to have the exact same RPC method signature defined in different builds and they are not necessarily going to be compatible with each other.
+
+## Deprecation of Return Values
+
+MLAPI supports RPC return values on convenience RPCs.
+
+Example:
+
+```cs
+public IEnumerator MyRpcCoroutine()
+{
+    RpcResponse<float> response = InvokeServerRpc(MyRpcWithReturnValue, Random.Range(0f, 100f), Random.Range(0f, 100f));
+
+    while (!response.IsDone)
+    {
+        yield return null;
+    }
+
+    Debug.LogFormat("The final result was {0}!", response.Value);
+}
+
+[ServerRPC]
+public float MyRpcWithReturnValue(float x, float y)
+{
+    return x * y;
+}
+```
+
+This RFC also drops this feature and similar functionality can be achived as following:
+
+```cs
+void MyRpcInvoker()
+{
+    MyRpcWithReturnValueRequestServerRpc(Random.Range(0f, 100f)), Random.Range(0f, 100f)));
+}
+
+[ServerRpc]
+void MyRpcWithReturnValueRequestServerRpc(float x, float y)
+{
+    MyRpcWithReturnValueResponseClientRpc(x * y);
+}
+
+[ClientRpc]
+void MyRpcWithReturnValueResponseClientRpc(float result)
+{
+    Debug.LogFormat("The final result was {0}!", result);
+}
+```
+
+## Tooling
+
+Tooling is one of the most important parts in overall developer experience, especially when we as framework developers are introducing a brand-new network construct, in this case RPCs.
+
+### Unity Editor Diagnostics
+
+In short, developers will see error and warning messages in the Unity Editor in the Console window if there was a problem with RPC method definition, parameters types on RPC method signature etc.
+
+Some of those messages will be warning messages which will not prevent compile and/or build.
+
+Some of those messages will be error messages which will prevent compile and/or build.
+
+Descriptive messages, message types, when messages should appear and other details are very much dependent on the implementation of this RPC API specification, therefore they will be decided while implementing the feature.
+
+### Roslyn Unity Analyzers
```

Comment:
Can we mention "Roslyn feature rollout will occur in sync with Roslyn availability in release Unity versions"

Reply:
Actually, Roslyn Analyzers are supported in Unity today (but Roslyn Source Generators is not). Also, in the "Strategy" section below, we're only highlighting Unity Diagnostics implementation and leaving both Rider Unity Plugin and Roslyn Unity Analyzers for the future and never promise a specific timeline or commitment.